### ISSUE TITLE
requests.utils function accepts invalid ip/cidr input

### ISSUE THREAD

**Issue by disconnect3d** [kennethreitz/requests#5131]
TLDR: On some libc implementations (e.g. glibc) the `socket.inet_aton` function parses IP strings trailed with whitespace and garbage.

An example below:
![image](https://user-images.githubusercontent.com/10009354/60627678-1f563a00-9def-11e9-88d3-3d37397164e5.png)

I have already filled an issue about this in `socket.inet_aton` but it is yet going to be discussed whether it will be fixed in Python.

## Expected Result

The `requests.utils` functions `address_in_network`, `is_ipv4_address` and `is_valid_cidr` should fail with invalid input.

## Actual Result

Invalid input is passed.

## Reproduction Steps

```python
import requests

print(requests.utils.address_in_network('1.1.1.1 wtf', '1.1.1.1/24'))
print(requests.utils.is_ipv4_address('1.1.1.1 disconnect3d was here...'))
print(requests.utils.is_valid_cidr('1.1.1.1 obviously not but yes/24'))
```

## System Information

The issue is related to libc implementation and has been tested on glibc 2.27 and 2.29.

It also occurs on MacOS which I am not sure if it is based on glibc.

**Comment by HJones82493** [comment]
PR #5135 
I feel as though the call to the socket library is actually unnecessary here.

**Comment by disconnect3d** [comment]
Ping @kennethreitz @Lukasa @sigmavirus24 @nateprewitt @slingamn can we finally get some love here and get this fixed?

This could probably also be marked as a security issue since if someone uses this in a security relevant context, they can get in trouble. (Although I doubt and hope that nobody does use those functions in such a way).

**Comment by sigmavirus24** [comment]
These are not intended for use by others. The fact the language has no way of making things as private or internal means that we can't stop people from giving a function not intended for public use garbage or getting garbage out

**Comment by sigmavirus24** [comment]
Also, @disconnect3d rather than demand free labor of others, you're welcome to submit a fix for this

**Comment by disconnect3d** [comment]
> These are not intended for use by others. The fact the language has no way of making things as private or internal means that we can't stop people from giving a function not intended for public use garbage or getting garbage out

The typical way to mark this is to prefix the name with underscore or two ;).

**Comment by sigmavirus24** [comment]
Yeah and people still use it from this library so we stopped bothering. I've been maintaining this library for almost a decade now. Writing python for longer. I know what the convention is and I know people don't care

**Comment by HJones82493** [comment]
Is there an argument against using `socket.inet_pton` instead of `socket.inet_aton`? It seems like `socket.inet_pton` has had its implementation fixed to not accept the extra strings and could solve this whole problem? If there's historical reason to not use it in these contexts, that's fine, but it seems like a simple fix (I'd be more than willing to submit something for review for it) if not.

**Comment by vcapparelli** [comment]
PR #6675 
@disconnect3d , the `socket` implementation was replaced by `ipaddress`, fixing the bugs

**Comment by disconnect3d** [comment]
Thanks! :)

**Comment by victorwss** [comment]
We already have a PR and I think that it is ok. Can it be merged? If not, why?
